dev/devicelab/bin/tasks/android_obfuscate_test.dart

```diff
@@ -38,20 +38,68 @@ Future<void> main() async {
         // Verify that an identifier from the Dart project code is not present
         // in the compiled binary.
         await inDirectory(flutterProject.rootPath, () async {
-          await exec('unzip', <String>[outputDirectory]);
+          await exec('unzip', <String>[outputApkDirectory]);
+          checkFileExists(path.join(flutterProject.rootPath, 'lib/armeabi-v7a/libapp.so'));
           final String response = await eval(
             'grep',
             <String>[flutterProject.name, 'lib/armeabi-v7a/libapp.so'],
             canFail: true,
           );
           if (response.trim().contains('matches')) {
-            foundProjectName = true;
+            foundApkProjectName = true;
           }
         });
       });
-      if (foundProjectName) {
-        return TaskResult.failure('Found project name in obfuscated dart library');
+
+      bool foundAarProjectName = false;
+      await runModuleProjectTest((FlutterModuleProject flutterProject) async {
+        section('AAR content with --obfuscate');
+
+        await inDirectory(flutterProject.rootPath, () async {
+          await flutter('build', options: <String>[
+            'aar',
+            '--target-platform=android-arm',
+            '--obfuscate',
+            '--split-debug-info=foo/',
+            '--no-debug',
+            '--no-profile',
+          ]);
+        });
+
+        final String outputAarDirectory = path.join(
+          flutterProject.rootPath,
+          'build/host/outputs/repo/com/example/aaa/flutter_release/1.0/flutter_release-1.0.aar',
+        );
+        final Iterable<String> aarFiles = await getFilesInAar(outputAarDirectory);
+
+        checkCollectionContains<String>(<String>[
+          ...flutterAssets,
+          'jni/armeabi-v7a/libapp.so',
+        ], aarFiles);
+
+        // Verify that an identifier from the Dart project code is not present
+        // in the compiled binary.
+        await inDirectory(flutterProject.rootPath, () async {
+          await exec('unzip', <String>[outputAarDirectory]);
+          checkFileExists(path.join(flutterProject.rootPath, 'jni/armeabi-v7a/libapp.so'));
+          final String response = await eval(
+            'grep',
+            <String>[flutterProject.name, 'jni/armeabi-v7a/libapp.so'],
+            canFail: true,
+          );
+          if (response.trim().contains('matches')) {
+            foundAarProjectName = true;
+          }
+        });
+      });
+
+      if (foundApkProjectName) {
+        return TaskResult.failure('Found project name in obfuscated APK dart library');
+      }
+      if (foundAarProjectName) {
+        return TaskResult.failure('Found project name in obfuscated AAR dart library');
```

Comment:
Bleh this fails:
```
[android_obfuscate_test] [STDOUT] Executing: grep aaa jni/armeabi-v7a/libapp.so in /var/folders/mv/qlhc4vsj6tqct166rw3zwmth00mfq2/T/flutter_devicelab_gradle_plugin_test.bY7bhQ/aaa
[android_obfuscate_test] [STDOUT] stdout: Binary file jni/armeabi-v7a/libapp.so matches
...
Task result:
{
  "success": false,
  "reason": "Found project name in obfuscated AAR dart library"
}

```

Reply:
Does the build aar call into assemble in the same way?